### PR TITLE
add scheduling framework extender

### PR DESCRIPTION
**Pull request by saintube**
Signed-off-by: saintube <saintube@foxmail.com>

### Ⅰ. Describe what this PR does

add a framework extender for the koord-scheduler, which enables us to extend the native `frameworkImpl` methods (`RunPreFilterPlugins`, `RunFilterPlugins`, ...)

### Ⅱ. Does this pull request fix one issue?

partially for #188 

### V. Checklist

- [X] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [X] All checks passed in `make test`

### CODE REVIEW

**Review by eahydra — COMMENTED** [review]
/lgtm

File: pkg/scheduler/frameworkext/framework_extender.go [hunk 1]
```diff
@@ -96,6 +99,96 @@ func (ext *frameworkExtendedHandleImpl) NodeResourceTopologySharedInformerFactor
...
+		if hooked {
+			klog.V(5).InfoS("RunPreFilterPlugins hooked", "meet PreFilterPhaseHook", "hook", hook.Name(), "pod", klog.KObj(pod))
+			return ext.Framework.RunPreFilterPlugins(ctx, cycleState, newPod)
+		}
+	}
+	return ext.Framework.RunPreFilterPlugins(ctx, cycleState, pod)
+}
+
+func (ext *frameworkExtenderImpl) RunFilterPlugins(ctx context.Context, cycleState *framework.CycleState, pod *corev1.Pod, nodeInfo *framework.NodeInfo) framework.PluginToStatus {
+	for _, hook := range ext.filterHooks {
```
Comment (buptcozy) [inline pkg/scheduler/frameworkext/framework_extender.go:181]: ext.filterHooks is a slice, but only execute first hoo, then return ext.Framework.RunFilterPlugins directly.  My question is that should ext.filterHooks be a slice or a single pointer? I prefer a slice, thus different plugin can has its own hook?

Comment (saintube) [reply pkg/scheduler/frameworkext/framework_extender.go:181]: Here extended RunFilterPlugins checks if any FilterHook confirms the hook and returns the native RunFilterPlugins.
For `Reservation`'s usage, this behavior is alright and has the same pattern with native `RunBindPlugins`.
We may standardize this framework extender later.

**Pull request merged**